docs/src/main/asciidoc/config-yaml.adoc

```diff
@@ -207,3 +207,58 @@ quarkus:
 
 YAML `null` keys are not included in the assembly of the configuration property name, allowing them to be used
 in any level for disambiguating configuration keys.
+
+== Current Limitation
+The smallrye-config library can only retrieve a single value at a time using a string representation of the path, and it does not provide a built-in way to retrieve complex nested structures in configuration files.  To work around this limitation, sources need to provide a single string representation of each value, effectively flattening the structure to a key-value pair of type `String -> String` . This means that nested structures in JSON or YAML files need to be represented in a flat format.
```

Comment:
As an extension user, there is no actionable item in the text. While this may be a limitation of the internal API, this is not a limitation for the user (only as an implementor, which is not the target of these docs). I'm not required to flatten my configuration structure to use it. In fact, the configuration example uses the expected JSON tree structure.

Reply:
fixed it